### PR TITLE
Display message if user is on a team that has submitted, but they were not a part of that team submission or any submission

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** Display message if user is on a team that has submitted, but they were not a part of that team submission or any submission

**What changed?**

- added function `does_team_have_submission` to `TeamMixin` that returns whether there is a team submission for the given team id for this block
- add a new template in `.../oa/response/` for the error state with the given error message
- call  `does_team_have_submission` in `get_submission_path_and_context` when user has no workflow, and if it is true, return the path for the new template

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5208](https://openedx.atlassian.net/browse/EDUCATOR-5208)

FIY: @edx/masters-devs-gta

Screen Shot 2020-08-26 at 11 47 47 AM

### CODE REVIEW

File: openassessment/xblock/submission_mixin.py [hunk 1]
```diff
@@ -783,11 +785,23 @@ def submission_path_and_context(self):
             context['save_status'] = self.save_status
             context['enable_delete_files'] = True
 
+            submit_enabled = True
```
Comment (jansenk) [inline openassessment/xblock/submission_mixin.py:788]: You might ask why I moved this, my answer is, why the heck did we add in the teams block in the middle of the function?????

Comment (muselesscreator) [reply openassessment/xblock/submission_mixin.py:788]: literally LOL

File: openassessment/xblock/submission_mixin.py [hunk 2]
```diff
@@ -783,11 +785,23 @@ def submission_path_and_context(self):
...
+            context['submit_enabled'] = submit_enabled
+
             if self.teams_enabled:
                 try:
                     team_info = self.get_team_info()
                     if team_info:
                         context.update(team_info)
+                        if self.does_team_have_submission(team_info['team_id']):
+                            no_workflow_path = 'openassessmentblock/response/oa_response_team_already_submitted.html'
                 except ObjectDoesNotExist:
```
Comment (muselesscreator) [inline openassessment/xblock/submission_mixin.py:805]: oh dear.  If mine goes in first, this will need to be moved again.

File: openassessment/xblock/submission_mixin.py [hunk 3]
```diff
@@ -783,11 +785,23 @@ def submission_path_and_context(self):
...
+                submit_enabled = False
+            context['submit_enabled'] = submit_enabled
+
             if self.teams_enabled:
                 try:
                     team_info = self.get_team_info()
                     if team_info:
                         context.update(team_info)
+                        if self.does_team_have_submission(team_info['team_id']):
+                            no_workflow_path = 'openassessmentblock/response/oa_response_team_already_submitted.html'
```
Comment (nsprenkle) [inline openassessment/xblock/submission_mixin.py:804]: Is there a reason to save this to `no_workflow_path` and not just update `path`?

File: openassessment/xblock/test/test_submission.py [hunk 4]
```diff
@@ -1328,6 +1328,57 @@ def test_change_team_with_submission(self, xblock):
         self.assertEqual(path, 'openassessmentblock/response/oa_response_submitted.html')
         self.assertEqual(context['student_submission'], create_submission_dict(individual_submission, xblock.prompts))
 
+    @scenario('data/team_submission.xml', user_id="Red Five")
+    def test_team_has_already_submitted(self, xblock):
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_submission.py:1332]: This is, visually, a beautiful test.

File: openassessment/xblock/test/test_submission.py [hunk 5]
```diff
@@ -1328,6 +1328,57 @@ def test_change_team_with_submission(self, xblock):
...
+            'test_course',
+            xblock.scope_ids.usage_id,
+            MOCK_TEAM_ID,
+            arbitrary_user.id,
+            ['rl', 'r2'],
+            {'text': 'This is the answer'},
+        )
+
+        # Assert that we return the 'team already submitted' path
+        self._assert_path_and_context(
```
Comment (nsprenkle) [inline openassessment/xblock/test/test_submission.py:1356]: Random thought: would love to see these `asserts` moved into a helper class and, more specifically here, have one for just `assert_path` because I don't know that we care about much of the context here.

File: openassessment/templates/openassessmentblock/response/oa_response_team_already_submitted.html [hunk 6]
```diff
@@ -0,0 +1,41 @@
+{% extends "openassessmentblock/response/oa_response.html" %}
+{% load i18n %}
+{% load tz %}
+
+{% block list_item %}
```
Comment (nsprenkle) [inline openassessment/templates/openassessmentblock/response/oa_response_team_already_submitted.html:5]: I like the `block` organization 👍

File: openassessment/xblock/submission_mixin.py (hunk 3 above)
Comment (jansenk) [reply openassessment/xblock/submission_mixin.py:804]: um .... now that I'm done with this, not really? at one point it was but then I moved something and it seems like it no longer is. (I'll stick it into our cleanup ticket so we can get these tickets closed)
